src/Navigation/Xamarin.Forms/Navigation.Xamarin.Forms/Service/NavigationService.show.cs

```diff
@@ -139,16 +120,46 @@ protected async Task<TViewModel> InitializeViewModelAsync<TViewModel>()
             return await InitializeViewModelInternal<TViewModel>();
         }
 
-        protected async Task<TViewModel> InitializeViewModelAsync<TViewModel, TModel>(
-            TModel model)
+        protected async Task<TViewModel> InitializeViewModelAsync<TViewModel, TData>(
+            TData data)
 
-            where TViewModel : class, IViewModel<TModel>
+            where TViewModel : class, IViewModel<TData>
         {
-            return await InitializeViewModelInternal<TViewModel, TModel>(
-                model);
+            return await InitializeViewModelInternal<TViewModel, TData>(
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Add '.ConfigureAwait(false)' to this call to allow execution to continue in any thread.](https://app.codacy.com/gh/UltimateCodeMonkeys/CodeMonkeysMVVM/pullRequest?prid=5598919)